packages/core/src/core/config/env.spec.ts

```diff
@@ -99,37 +106,106 @@ describe('Env', () => {
       testConfigFile('.env.test', 'test');
     });
 
+    context('given NODE_ENV is defined and .env.${NODE_ENV}.local exists', () => {
+      testConfigFile('.env.test.local', 'test');
+    });
+
     context('given NODE_ENV is not defined and .env.development exists', () => {
       testConfigFile('.env.development');
     });
 
+    context('given NODE_ENV is not defined and .env.development.local exists', () => {
+      testConfigFile('.env.development.local');
+    });
+
     context('given a .env file exists', () => {
       testConfigFile('.env');
     });
 
+    context('given a .env.local file exists', () => {
+      testConfigFile('.env.local');
+    });
+
     context('given multiple config files exist', () => {
 
       afterEach(() => {
         removeFile('.env');
+        removeFile('.env.local');
         removeFile('.env.development');
+        removeFile('.env.development.local');
       });
 
       describe('should return the configuration value', () => {
 
-        it('with default.yml overriding default.js', () => {
+        it('and determine the variable values, based on configuration file importance', () => {
```

Comment:
The tests added below don't test fully the order in which files are loaded. For example, if lines 17 and 18 or lines 18 and 19 are interchanged, the tests still pass whereas they shouldn't.

I think we could go simpler here by just extending the first test as follows. This would probably reduce the amount of code to maintain.
```typescript
const dotEnvContent2 = `
FOO_BAR=hello2
FOO_BAR_THREE=two
`;
const dotEnvContent3 = `
FOO_BAR_THREE=three
FOO_BAR_FOUR=three
`;
const dotEnvContent4 = `
FOO_BAR_FOUR=four
`;

it('based on configuration file importance.', () => {
  writeFileSync('.env.development.local', dotEnvContent4);
  writeFileSync('.env.development', dotEnvContent3);
  writeFileSync('.env.local', dotEnvContent2);
  writeFileSync('.env', dotEnvContent);

  strictEqual(Env.get('FOO_BAR_FOUR'), 'four');
  strictEqual(Env.get('FOO_BAR_THREE'), 'three');
  strictEqual(Env.get('FOO_BAR'), 'hello2');
});
```